src/pages/PageProfile.vue

```diff
@@ -25,36 +24,62 @@
 </template>
 
 <script>
+import EnvironmentMixin from '@/components/_mixins/EnvironmentMixin.vue';
 import Error from '@/components/ui/Error.vue';
 import Profile from '@/components/profile/Profile.vue';
 import LoadingSpinner from '@/components/ui/LoadingSpinner.vue';
 import { DISPLAY_PROFILE } from '@/queries/profile';
 import Related from '@/assets/js/related';
 import Page404 from './PageUnknown.vue';
+import PreviewAs from '@/components/profile/PreviewAs.vue';
 
 export default {
   name: 'PageProfile',
   components: {
     Error,
     Profile,
     Page404,
+    PreviewAs,
     LoadingSpinner,
   },
+  mixins: [EnvironmentMixin],
   computed: {
     variables() {
       if (
         this.$route.params.username ===
         this.$store.state.user.primaryUsername.value
       ) {
-        return { username: null };
+        return {
```

Comment:
It seems really messy to me to add an extra data binding to each element that we pass down to each element from this one place. It also seems like the source of this "viewas" functionality doesn't really belong in the PageProfile component. That's more of an opinion that I'm willing to change my mind on, but I think this might belong in it's own mixin.